### PR TITLE
phonzammi/error-handling-for-server-routing

### PR DESCRIPTION
**Pull request by phonzammi**
Hi @brillout and @AurelienLourot 

What is inside this PR ?
1. Sync dependencies
- Remove "@vue/compiler-sfc" and "@vue/server-renderer". [The Reasons](https://github.com/vuejs/core/tree/main/packages/server-renderer#vueserver-renderer)
- Downgrade "vite": "^5.0.7", to "vite": "^4.5.1",
- Up vike to "vike": "^0.4.150",
- Up vue to "vue": "^3.3.12"
- Rebase lockfile, please take a look what is changed to see if something important is missing 🙏

2. Workarounds for error handling (renderToString & renderToNodeStream)
This is meant for server routing since onRenderHtml only run with server routing (on the first page visit or when page reload). 
Whereas for client routing (especially for pages with `<script setup>` or Composition API), I think we let the user (developer) create their own error handler, e.g. with Error Boundary. Even they can choose to replace `/pages/_error/+Page.tsx` with custom Error Boundary when the page reload or not (I've tested both).

3. Minor change ssr to vike inside vite.config.ts

As always, my PR is far from perfect, please review and improve this PR for the best of vike-vue.

### CODE REVIEW

File: vike-vue/package.json [hunk 1]
```diff
@@ -17,18 +17,16 @@
...
   },
   "peerDependencies": {
-    "@vitejs/plugin-vue": "^4.0.0",
-    "@vue/compiler-sfc": "^3.2.47",
-    "@vue/server-renderer": "^3.2.47",
-    "vite": "^4.4.9",
-    "vike": "^0.4.147",
-    "vue": "^3.3.4"
+    "@vitejs/plugin-vue": "^4.5.2",
+    "vite": "^4.5.1",
```
Comment (AurelienLourot) [inline vike-vue/package.json:21]: Should it be something like `^4.5.1 || ^5.0.7` so we support both Vite 4 and 5?

And then we wouldn't need to downgrade Vite in the examples?

**Review by AurelienLourot — CHANGES_REQUESTED** [review]
LGTM, thanks a lot for doing this :pray: Just one tiny question

File: vike-vue/package.json (hunk 1 above)
Comment (phonzammi) [reply vike-vue/package.json]: Done

**Review by AurelienLourot — APPROVED** [review]
Perfect, thanks a lot!

**Comment by brillout** [comment]
> phonzammi requested review from brillout 

👍 LGTM as well. One thing: I think there is potential that we're swallowing errors. I'd be extra careful that errors emitted by `app.config.errorHandler` are always shown to the user. E.g. what happens if an error is thrown while streaming? I think it will be swallowed.

**Comment by AurelienLourot** [comment]
@brillout not sure I get it. Are you saying this patch introduces a problem or are you saying it doesn't cover everything yet?

**Pull request merged**